src/nft/components/details/detailsV2/DataPageTraits.tsx

```diff
@@ -1,23 +1,101 @@
 import { Trans } from '@lingui/macro'
-import { GenieAsset } from 'nft/types'
+import Column from 'components/Column'
+import { ScrollBarStyles } from 'components/Common'
+import Row from 'components/Row'
+import { useSubscribeScrollState } from 'nft/hooks'
+import { Trait } from 'nft/types'
 import { useMemo } from 'react'
 import styled from 'styled-components/macro'
+import { ThemedText } from 'theme'
+import { opacify } from 'theme/utils'
 
 import { Tab, TabbedComponent } from './TabbedComponent'
+import { TraitRow } from './TraitRow'
 
-const TraitsContentContainer = styled.div`
-  height: 492px;
+const TraitsHeaderContainer = styled(Row)`
+  padding-right: 12px;
 `
 
-const TraitsContent = () => {
-  return <TraitsContentContainer>Traits Content</TraitsContentContainer>
+const TraitsHeader = styled(ThemedText.SubHeaderSmall)<{ $flex?: number; justify?: string }>`
+  display: flex;
+  line-height: 20px;
+  color: ${({ theme }) => theme.textSecondary};
+  flex: ${({ $flex }) => $flex ?? 1};
+  justify-content: ${({ justify }) => justify};
+`
+
+const TraitRowContainer = styled.div`
+  position: relative;
+`
+
+const TraitRowScrollableContainer = styled.div`
+  overflow-y: auto;
+  overflow-x: hidden;
+  max-height: 412px;
+  width: calc(100% + 6px);
+
+  ${ScrollBarStyles}
+`
+
+const Scrim = styled.div<{ isBottom?: boolean }>`
+  position: absolute;
+  height: 88px;
+  left: 0px;
+  right: 6px;
+
+  ${({ isBottom }) =>
+    isBottom
+      ? 'bottom: 0px'
+      : `
+        top: 0px;
+        transform: matrix(1, 0, 0, -1, 0, 0);
+      `};
+
+  background: ${({ theme }) =>
+    `linear-gradient(180deg, ${opacify(0, theme.backgroundSurface)} 0%, ${theme.backgroundSurface} 100%)`};
+  display: flex;
+`
+
+const TraitsContent = ({ traits }: { traits?: Trait[] }) => {
+  const { userCanScroll, scrollRef, scrollProgress, scrollHandler } = useSubscribeScrollState()
+
+  // This is needed to prevent rerenders when handling scrolls
+  const traitRows = useMemo(() => {
+    return traits?.map((trait) => <TraitRow trait={trait} key={trait.trait_type + ':' + trait.trait_value} />)
+  }, [traits])
+
+  return (
+    <Column>
+      <TraitsHeaderContainer>
+        <TraitsHeader $flex={3}>
```

Comment:
nit: when did we start using `$` for variables? (I know the way $ works with Styled components). What makes `flex` different from e.g. `justify` (used elsewhere for TraitsHeader)

Reply:
We use `$` when the param name matches an exact css style such as `$flex` or `$color`